tests/structures/logic/test_z3_logic_converter.py

```diff
@@ -0,0 +1,55 @@
+import pytest as pytest
+from decompiler.structures.pseudo import BinaryOperation, Branch, Condition, Constant, OperationType, Variable
+from decompiler.structures.pseudo.logic import BaseConverter
+from decompiler.structures.pseudo.z3_logic import Z3Converter
+from z3 import BoolRef
+
+
+def _generate_instr_bool_as_numbers(op: OperationType) -> Branch:
+    return Branch(
+        Condition(
+            OperationType.not_equal,
+            [
+                BinaryOperation(
+                    op,
+                    [
+                        Condition(OperationType.greater_us, [Variable("a"), Constant(64)]),
+                        Condition(OperationType.less_or_equal_us, [Variable("b"), Constant(63)]),
+                    ],
+                ),
+                Constant(-1),
+            ],
+        )
+    )
+
+
+@pytest.mark.parametrize(
+    "instr",
+    [
+        _generate_instr_bool_as_numbers(OperationType.minus),
+        _generate_instr_bool_as_numbers(OperationType.plus),
+        _generate_instr_bool_as_numbers(OperationType.multiply),
+        _generate_instr_bool_as_numbers(OperationType.divide),
+        _generate_instr_bool_as_numbers(OperationType.left_shift),
+        _generate_instr_bool_as_numbers(OperationType.right_shift),
+        _generate_instr_bool_as_numbers(OperationType.less),
+        _generate_instr_bool_as_numbers(OperationType.less_or_equal),
+        _generate_instr_bool_as_numbers(OperationType.greater),
+        _generate_instr_bool_as_numbers(OperationType.greater_or_equal),
+        _generate_instr_bool_as_numbers(OperationType.modulo),
+    ],
+)
```

Comment:
I think a few are missing (see above)

Reply:
I only tested the operation types which I fixed, but will add the other ones as well.